### PR TITLE
Samples: Bluetooth: CAP Initiator broadcast support 

### PR DESCRIPTION
**Pull request by Thalley**
Add broadcast support to the CAP initiator sample.

This adds new sample-specific Kconfig options to help
select the right Kconfig options based on whether
unicast, broadcast or both is being used.

This also moves common TX functionality
to cap_initiator_tx to reuse the same TX thread
and functionality.

Finally there is a babblesim implemented for
the broadcast. There is not broadcast support for the
CAP acceptor sample yet, so this test only verifies that we
get the TX complete events from the controller.

### CODE REVIEW

**Comment by Thalley** [comment]
CI is failing due to this bug: https://github.com/zephyrproject-rtos/zephyr/pull/74299

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased without changes

**Review by fredrikdanebjer — COMMENTED** [review]
Overall, lgtm. I have two very minor comments regarding your comments I'm afraid :)

File: samples/bluetooth/cap_initiator/README.rst [hunk 1]
```diff
@@ -2,13 +2,15 @@
    :name: Bluetooth: Common Audio Profile Initiator
    :relevant-api: bt_cap bt_bap bluetooth
 
-   CAP Initiator sample that connects to CAP Acceptors and setup audio streaming.
+   CAP Initiator sample that connects to CAP Acceptors and setup audio streaming,
```
[the remark this answers is not in the document]
Comment (Thalley) [reply samples/bluetooth/cap_initiator/README.rst]: Good catch. I've made it more clear when there is talk about unicast and broadcast audio streams.

**Review by kruithofa — APPROVED** [review]

**Review by fredrikdanebjer — APPROVED** [review]

**Pull request merged**